### PR TITLE
add rego support

### PR DESCRIPTION
**Pull request by rzjfr**
Where are the tests? Have you added tests? Have you updated the tests? 

```
test/fixers/test_opa_fmt_fixer_callback.vader
test/linter/test_rego_opacheck.vader

```

Related issue:

https://github.com/dense-analysis/ale/issues/2531

### CODE REVIEW

File: doc/ale.txt [hunk 1]
```diff
@@ -3126,6 +3126,9 @@ documented in additional help files.
     ols...................................|ale-reasonml-ols|
     reason-language-server................|ale-reasonml-language-server|
     refmt.................................|ale-reasonml-refmt|
+  rego....................................|ale-rego-options|
```
Comment (hsanson) [inline doc/ale.txt:3129]: Since you added cpell support would be nice to have cpell added here. Same for the other index documents.

**Review by hsanson — CHANGES_REQUESTED** [review]
Looks good and complete. Just missing cspell support in the list of supported tools for rego file.

File: doc/ale.txt (hunk 1 above)
Comment (rzjfr) [reply doc/ale.txt:3129]: @hsanson thanks for reviewing this PR, good catch, I've added the required docs now.

**Review by hsanson — APPROVED** [review]
Thanks for the prompt fixes.

**Pull request merged**